File: Lab9/app/src/main/java/com/example/lab9/ListAdapter.java
package com.example.lab9;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

public class ListAdapter extends RecyclerView.Adapter<ElemViewHolder> {
    private int countItems;

    private List<TableEntity> itemsList;

    public ListAdapter(List<TableEntity> itemsList) {
        countItems = 0;
        this.itemsList = itemsList;
    }

    @NonNull
    @Override
    public ElemViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        Context context = parent.getContext();
        int layoutIdForListItem = R.layout.list_item;
        LayoutInflater inflater = LayoutInflater.from(context);
        boolean shouldAttachToParentImmediately = false;

        View view = inflater.inflate(layoutIdForListItem, parent, shouldAttachToParentImmediately);
        ElemViewHolder viewHolder = new ElemViewHolder(view, "");

        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull final ElemViewHolder holder, int position) {
        // Обновляем элемент
        holder.updateElem(itemsList.get(position).id, itemsList.get(position).text);
    }

    @Override
    public int getItemCount() {
        if (itemsList == null) {
            return 0;
        }
        return itemsList.size();
    }

    public void deleteElem(long i) {
        itemsList.remove(i);
        notifyDataSetChanged();
    }

    public void clearData() {
        if (itemsList != null) {
            itemsList.clear();
        }
        notifyDataSetChanged();
    }

    public void setData(List<TableEntity> elems) {
        itemsList = elems;
        notifyDataSetChanged();
    }
}

File: Lab7/app/src/main/java/lab7/ListAdapter.kt
package com.example.koshelev.labs.javalab7

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

class ListAdapter(private var itemsList: MutableList<TableEntity>?) : RecyclerView.Adapter<ElemViewHolder>() {
    private val countItems: Int

    init {
        countItems = 0
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ElemViewHolder {
        val context = parent.context
        val layoutIdForListItem = R.layout.list_item
        val inflater = LayoutInflater.from(context)
        val shouldAttachToParentImmediately = false

        val view = inflater.inflate(layoutIdForListItem, parent, shouldAttachToParentImmediately)

        return ElemViewHolder(view, "")
    }

    override fun onBindViewHolder(holder: ElemViewHolder, position: Int) {
        // Обновляем элемент
        holder.setDataAndUpdate(itemsList!![position].text)
    }

    override fun getItemCount(): Int {
        return if (itemsList == null) {
            0
        } else itemsList!!.size
    }

    fun clearData() {
        if (itemsList != null) {
            itemsList!!.clear()
        }
        notifyDataSetChanged()
    }

    fun setData(elems: MutableList<TableEntity>) {
        itemsList = elems
        notifyDataSetChanged()
    }
}
